### PR TITLE
Api for opponents listing

### PR DESCRIPTION
**Pull request by MayurDeshmukh10**

### CODE REVIEW

**Comment by MayurDeshmukh10** [comment]
[Opponents Listing API](https://trello.com/c/2Ms8Bx9U/45-opponents-listing-api)

File: .rubocop.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ Layout/LineLength:
  Max: 125
 
 Metrics/MethodLength:
- Max: 15
+ Max: 35
```
Comment (tanya-saroha) [inline .rubocop.yml]: @MayurDeshmukh10  Why 35? Method length should never be this much!

File: spec/requests/post_opponents_listing_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  context "returns error" do
+    it "returns unauthorized error due to user not present" do
+      @token2 = JsonWebToken.encode(user_id: 100)
+      get "/api/v1/list_opponents", headers: {"user-auth-token" => @token2}
+      expect(response). to have_http_status(:unauthorized)
+    end
+
+    it "returns unauthorized error due to JWT token error" do
+      @token3 = "eyJhbGciOiJIRVyX2lkIjoiMSJ9.O5Y5s_hvEW8BM7E8jq6HihxQ0DDFxO_2_xtnrvVj4PY"
```
Comment (tanya-saroha) [inline spec/requests/post_opponents_listing_spec.rb]: @MayurDeshmukh10 Dont pass the hard coded token. Generate it. You can write a method and call it everywhere in your test cases

File: .rubocop.yml (hunk 1 above)
Comment (MayurDeshmukh10) [reply .rubocop.yml]: Changed method length to 15

File: spec/requests/post_opponents_listing_spec.rb (hunk 2 above)
Comment (MayurDeshmukh10) [reply spec/requests/post_opponents_listing_spec.rb]: Token here is a invalid token with invalid signature. So cannot be generated by method. I have generated other tokens using a method.

File: config/locales/en.yml [hunk 3]
```diff
@@ -31,6 +31,9 @@
 
 en:
   hello: "Hello world"
+  list_opponents:
+    id_not_found: "Id not found"
```
[the remark this answers is not in the document]
Comment (MayurDeshmukh10) [reply config/locales/en.yml]: Message corrected

**Comment by tanya-saroha** [comment]
@Ashviniv please merge this PR if everything is fine

**Pull request merged**